KPI/H.adoc

```diff
@@ -1,99 +1,237 @@
-       </gmd:type>
-       <gmd:thesaurus>
-          <gmd:CI_Citation>
-             <gmd:title>
-                <gco:CharacterString>WMO Codes</gco:CharacterString>
-             </gmd:title>
-              <gmd:date>
-                 <gmd:CI_Date>
-                    <gmd:date>
-                       <gco:Date>2016-06-25</gco:Date>
-                    </gmd:date>
-                 <gmd:dateType>
-                    <gmd:CI_DateTypeCode codeList="http://www.isotc211.org/2005/resources/Codelist/gmxCodelists.xml#CI_DateTypeCode" codeListValue="revision">revision</gmd:CI_DateTypeCode>
-                 </gmd:dateType>
-              </gmd:CI_Date>
-           </gmd:date>
-       </gmd:CI_Citation>
-    </gmd:thesaurus>
+  <gmd:MD_Keywords>
+    <gmd:thesaurusName>
+      <gmd:CI_Citation>
+        <gmd:title>
+          <gmx:Anchor xlink:href="http://codes.wmo.int/grib2/codeflag/_4.2">WMO Codes Registry GRIB edition 2</gmx:Anchor>
+        </gmd:title>
+        <gmd:date>
+          <gmd:CI_Date>
+            <gmd:date>
+              <gco:Date>2016-06-25</gco:Date>
+            </gmd:date>
+            <gmd:dateType>
+              <gmd:CI_DateTypeCode codeList="http://standards.iso.org/iso/19139/resources/gmxCodelists.xml#CI_DateTypeCode" codeListValue="publication">publication</gmd:CI_DateTypeCode>
+            </gmd:dateType>
+          </gmd:CI_Date>
+        </gmd:date>
+      </gmd:CI_Citation>
+    </gmd:thesaurusName>
   </gmd:MD_Keywords>
-</gmd:descriptiveKeywords>
-....
+<gmd:descriptiveKeywords>
+```
+
+Check for referenced thesaurus via `gco:CharacterString` elements (use of a URL
+within `gco:CharacterString` is not recommended, `gmx:Anchor` should be used
+instead).
+
+```xml
+<gmd:descriptiveKeywords>
+  <gmd:MD_Keywords>
+    <gmd:thesaurusName>
+      <gmd:CI_Citation>
+        <gmd:title>
+          <gco:CharacterString>WMO Codes Registry - GRIB edition 2</gco:CharacterString>
+        </gmd:title>
+        <gmd:date>
+          <gmd:CI_Date>
+            <gmd:date>
+              <gco:Date>2016-06-25</gco:Date>
+            </gmd:date>
+            <gmd:dateType>
+              <gmd:CI_DateTypeCode codeList="http://standards.iso.org/iso/19139/resources/gmxCodelists.xml#CI_DateTypeCode" codeListValue="publication">publication</gmd:CI_DateTypeCode>
+            </gmd:dateType>
+          </gmd:CI_Date>
+        </gmd:date>
+      </gmd:CI_Citation>
+    </gmd:thesaurusName>
+  </gmd:MD_Keywords>
+<gmd:descriptiveKeywords>
+```
+
+.Keywords implementation rules
+|===
+|Rule |Score
+
+a|`gmd:keywords` are user-defined. There is no presence of `gmd:type` or
+`gmd:thesaurusName`
+|1
 
-=== H.2 What is being measured ?
+|`gmd:keywords`, `gmd:type`, and `gmd:thesaurusName` are defined. Terms are
+defined by `gco:CharacterString` elements, not URIs via `gmx:Anchor`
+|1
 
-The presence of well defined data parameters, from well known thesaurus.
+|Compliance rules are implemented and they conform to well defined keywords rules.
```

Comment:
I guess compliance rules are being checked out in A (presence)
and with this one we are checking they follow the  expected format.

Reply:
Correct, given KPI A is passed, and KPI H compliance rules are met, then 2 points are added to the score.